Lib/test/test_typing.py

```diff
@@ -5004,6 +5004,18 @@ def test_args_kwargs(self):
         self.assertEqual(repr(P.args), "P.args")
         self.assertEqual(repr(P.kwargs), "P.kwargs")
 
+    def test_stringized(self):
+        P = ParamSpec('P')
+        class C(Generic[P]):
+            func: Callable["P", int]
+            def foo(self, *args: "P.args", **kwargs: "P.kwargs"):
```

Comment:
I don't think that this a reasonable thing to write, but I don't object either.

You can add this class to a module with `from __future__ import annotations`, I guess it makes more sense.

See [`test.ann_module6`](https://github.com/python/cpython/blob/main/Lib/test/ann_module5.py) as an example

Reply:
Maybe we should do both? I think if stringifcation outlives PEP 563 (a real possibility, it seems), then it will be nice to have these cases around. But I'm also hesitant to include a ton of files if we start adding these test cases for other typeforms.

Reply:
> But I'm also hesitant to add a ton of files if we start adding these test cases for other typeforms.

We can only one: `ann_future` and use it for all valid `__future__` annotations.

Reply:
I like that a lot. I would support a PR making that change, if you want it. This PR is going to need to wait for discussions to play out for a while yet, so such a change might be able to get merged first.

Reply:
Ok 🙂